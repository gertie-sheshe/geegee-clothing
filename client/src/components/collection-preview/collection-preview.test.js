import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from './collection-preview.component';
import CollectionItem from '../collection-item/collection-item.component';

describe('Collection Preview Tests', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  let mockItems = [{ id: 1 }, { id: 2 }];

  const mockRouteName = 'hats';

  beforeEach(() => {
    mockMatch = {
      path: '/shop',
    };

    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      routeName: mockRouteName,
      title: 'hats',
      items: mockItems,
    };

    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  it('Should render Collection Preview Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render Collection Item component based on the number of items', () => {
    expect(wrapper.find(CollectionItem).length).toEqual(mockItems.length);
  });
  // Uncomment this test later
  //   it('Should call history.push with the right string when TitleContainer clicked', () => {
  //     wrapper.find('TitleContainer').simulate('click');
  //     expect(mockHistory.push).toHaveBeenCalledWith(
  //       `${mockMatch.path}/${mockRouteName}`,
  //     );
  //   });
});
