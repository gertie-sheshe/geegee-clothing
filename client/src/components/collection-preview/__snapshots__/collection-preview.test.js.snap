// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Preview Tests Should render Collection Preview Component 1`] = `ShallowWrapper {}`;
